bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -596,20 +593,13 @@ impl Timeline {
     /// being sent already. If the event was a remote event, then it will be
     /// redacted by sending a redaction request to the server.
     ///
-    /// Returns whether the redaction did happen. It can only return false for
-    /// local events that are being processed.
+    /// Will return an error if the event couldn't be redacted.
     pub async fn redact_event(
         &self,
-        item: Arc<EventTimelineItem>,
+        id: String,
```

Comment:
See https://github.com/matrix-org/matrix-rust-sdk/pull/3942#discussion_r1768852442.

It can be both the event id or a transaction id. Maybe I should rename the function to `redact_by_event_id` and `redact_by_transaction_id`, or use some wrapper. I listed the possible implementations I thought about in that comment, but I don't really know which one is best.